portal/v2/src/ClusterDetailListComponents/Statistics/StatisticsComponent.tsx

```diff
@@ -0,0 +1,165 @@
+  width: number,
+  endDate: Date,
+}) {
+  const width = props.width
+  const height = props.height
+  const [points, setPoints] = useState<ILineChartPoints[]>([])
+  const [data, setData] = useState<IChartProps>({})
+  const [spinnerVisible, setSpinnerVisible] = useState<boolean>(true)
+  const timeFormat = '%H:%M'
+
+  const colors: string[] = [
+    DefaultPalette.blue,
+    DefaultPalette.blueLight,
+    DefaultPalette.blueDark,
+    DefaultPalette.blueMid,
+    DefaultPalette.black,
+    DefaultPalette.red,
+    DefaultPalette.redDark,
+    DefaultPalette.yellow,
+    DefaultPalette.yellowDark,
+    DefaultPalette.yellowLight,
+    DefaultPalette.green,
+    DefaultPalette.greenLight,
+    DefaultPalette.greenDark,
+    DefaultPalette.purple,
+    DefaultPalette.purpleLight,
+    DefaultPalette.purpleDark,
+    DefaultPalette.orange,
+    DefaultPalette.orangeLight,
+    DefaultPalette.orangeLighter,
+    DefaultPalette.magenta,
+    DefaultPalette.magentaDark,
+    DefaultPalette.magentaLight,
+    DefaultPalette.themePrimary,
+    DefaultPalette.neutralPrimary,
+    DefaultPalette.neutralDark,
+    DefaultPalette.neutralSecondary,
+    DefaultPalette.neutralTertiary,
+    DefaultPalette.teal,
+    DefaultPalette.tealDark,
+    DefaultPalette.tealLight,
+    DefaultPalette.accent,
+    DefaultPalette.themeDarker,
+    DefaultPalette.themeDarkAlt,
+    DefaultPalette.themeDark,
+    DefaultPalette.themeLight,
+    DefaultPalette.themeLighter,
+    DefaultPalette.themeLighterAlt,
+    DefaultPalette.themePrimary,
+    DefaultPalette.themeSecondary,
+    DefaultPalette.themeTertiary
+  ]
+
+  const _onLegendClickHandler = (selectedLegend: string | null | string[]): void => {
+    if (selectedLegend !== null) {
+      console.log(`Selected legend - ${selectedLegend}`);
+    }
+  };
+
+  function StatisticsHelperComponent(): JSX.Element {   
+    useEffect(() => {
+      if (props.metrics.length > 0) {
+        const newPoints: ILineChartPoints[] = []
+        props.metrics.forEach((metric, i) => {          
+          var dataPoints: ILineChartDataPoint[] = []
+          metric.MetricValue.forEach(metricValue => {
+            let timeStamp = new Date(metricValue.timestamp)
+            let metricsTime = convertToUTC(timeStamp)
+            var data: ILineChartDataPoint = {
+              x: metricsTime, y: metricValue.value
+            }
+            dataPoints.push(data)
+          })
+
+          var lineChartPoint: ILineChartPoints = {
+            legend: metric.Name,
+            onLegendClick: _onLegendClickHandler,
+            data: dataPoints,
+            color: colors[i]
+          }
+          newPoints.push(lineChartPoint)
+        })
+        setPoints(newPoints)    
+      }
+    },[props.metrics])
+
+    useEffect(() => {   
+      setData({
+        chartTitle: 'Line Chart',
+        lineChartData: points,
+      });
+      (points.length > 0) ? setSpinnerVisible(false) : setSpinnerVisible(true)
+    }, [points])
```

Comment:
These use effects don't appear to me to be needed. If the props are updated the component will be re-rendered.

Reply:
It is required, because, the useEffect just above this one updates the `points` but in react the state does not get updated immediately and hence if the put the logic of the useEffect in question to the above useEffect then, there would be no data for the graph to show, it would be spinner only, hence I have to set the data to be given to graph when the points get updated and then, there would be data for the graph.